### DISCUSSION TITLE
What's the reasoning that the template examples do not link Static Web Apps to APIM?

### DISCUSSION THREAD

**Discussion by ChristopherL-STCU** [Azure/azure-dev#1469] [Q&A]
In the various templates, for example [todo-csharp-cosmos-sql](https://github.com/Azure-Samples/todo-csharp-cosmos-sql), the Static Web App URL is [added as an _origin_ to the APIM policy](https://github.com/Azure-Samples/todo-csharp-cosmos-sql/blob/ee35d1f6a562fac94f803bac099fad505c0f4b76/infra/app/apim-api-policy.xml#LL8C18-L8C24) instead of [linking the Static Web App to the APIM](https://learn.microsoft.com/en-us/azure/static-web-apps/apis-api-management). Is there a reason linking was not used? It seems better to me to link, otherwise all of the API calls in the Static Web App will need to be updated to call the APIM, correct?

I'm working with a Blazor WASM application deployed as a Static Web App. I'm thinking the only way it will work with APIM is by linking the Static Web App to APIM. Is that also correct?

**Comment by rajeshkamal5050** [comment]
@vieiraae can you take a look at this and suggest what's the best practice to configure SWA <-> APIM?

cc: @jongio @weikanglim

**Comment by ChristopherL-STCU** [comment]
[I linked a SWA to APIM in a proof of concept](https://github.com/ciwchris/ShoppingList/tree/main/infra). It seems to work well except [when I try using auth in SWA then the API breaks](https://github.com/Azure/static-web-apps/discussions/1065), the API returns a HTTP status code 500. I haven't researched this issue in depth yet. I suspect it's because I'm not understanding how the SWA auth works with APIM. It works fine when only using the subscription keys, i.e. without using SWA auth. But when including SWA auth something else must change which breaks it.

**Reply by ChristopherL-STCU** [reply]
I determined the HTTP 500 occurs when using the APIM Consumption SKU but not when using the Developer SKU. There must be a limitation when linking a SWA to a Consumption APIM and when using the built-in SWA auth.
